db/migrate/20160926083857_create_donations.rb

```diff
@@ -0,0 +1,12 @@
+class CreateDonations < ActiveRecord::Migration
+  def change
+    create_table :donations do |t|
+      t.references :donor, index: true, foreign_key: true
+      t.references :cause, index: true, foreign_key: true
+      t.decimal :amount
```

Comment:
We are validating presence in our model, so we should add `null: false` here as well. That prevents invalid amounts to be entered into the database from other sources:

```
t.decimal :amount, null: false
```